### DISCUSSION TITLE
Issues upgrading from 0.16.1 to 0.17.3

### DISCUSSION THREAD

**Discussion by mhratson** [vectordotdev/vector#9793] [Q&A] [answered]
# Issues encountered 

They weren't mentioned in upgrade guide so i'm documenting it here for there record.

## 1 Undocumented Rename 

Before we had `rtt_threshold_ratio` in our `sinks.datadog-metrics` sink. After we're getting error:

 ```
    Running tests
    Oct 25 23:58:28.508 ERROR vector::unit_test: Failed to execute tests:
    unknown field `rtt_threshold_ratio`, expected one of `decrease_ratio`, `ewma_alpha`, `rtt_deviation_scale` for key `sinks.datadog-metrics` at line 45 column 1.
```

Solution: to use `rtt_deviation_scale` instead.

## 2 `nil` is no longer accepted as VRL

     error[E701]: call to undefined variable
         ┌─ :10:14
         │
      10 │    if err == nil {
         │              ^^^
         │              │
         │              undefined variable
         │              did you mean "err"?
         │
         = see language documentation at https://vrl.dev

Solution: use `null` instead.

**Comment by bruceg** [comment, marked as answer]
1. That is correct, and we missed that in the upgrade notes.
2. This one is a bit of an accident. `nil` was never valid VRL. However it was previously parsed as an undeclared variable, and all such variables contained the value `null`. In 0.17, VRL now checks for undeclared variables and produces an error.
